conduit/init_provider.go

```diff
@@ -0,0 +1,27 @@
+package conduit
+
+import (
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+)
+
+// PipelineInitProvider algod based init provider
+type PipelineInitProvider struct {
+	currentRound basics.Round
+	genesis      *bookkeeping.Genesis
+}
+
+// AdvanceDBRound advances the database round
+func (a *PipelineInitProvider) AdvanceDBRound() {
+	a.currentRound = a.currentRound + 1
+}
```

Comment:
I'm kind of skeptical that this is a good way to manage round state in the pipeline, but I'm interested to see what you have in mind in the next PR.